File: elixir/numbers/test/numbers_iterative_test.exs
defmodule NumbersIterativeTest do
  use ExUnit.Case
  doctest NumbersIterative

  test "fibonacci of 10" do
    assert NumbersIterative.fibonacci(10) == 55
  end

  test "factorial of 5" do
    assert NumbersIterative.factorial(5) == 120
  end

  test "sum of first 5 numbers" do
    assert NumbersIterative.sum_numbers(5) == 15
  end
end

File: julia/numbers/test/numbers_iterative_test.jl
#!/usr/bin/julia

using Test
include("../src/NumbersIterative.jl")
using .NumbersIterative

@testset "Iterative Tests" begin
  @testset "Fibonacci Tests" begin
    @test fibonacci(5) == 5
  end

  @testset "Factorial Tests" begin
    @test factorial(5) == 120
  end

  @testset "Sum Numbers Tests" begin
    @test sum_numbers(5) == 15
  end
end